src/typeahead/typeahead-window.ts

```diff
@@ -69,16 +77,22 @@ export class NgbTypeaheadWindow implements OnInit {
    */
   @Output('select') selectEvent = new EventEmitter();
 
+  @Output('activeChanged') activeChangedEvent = new EventEmitter();
+
   getActive() { return this.results[this.activeIdx]; }
 
-  markActive(activeIdx: number) { this.activeIdx = activeIdx; }
+  markActive(activeIdx: number) {
+    this.activeIdx = activeIdx;
+    this.activeChangedEvent.emit(this.id + '-' + this.activeIdx);
+  }
 
   next() {
     if (this.activeIdx === this.results.length - 1) {
       this.activeIdx = this.focusFirst ? (this.activeIdx + 1) % this.results.length : -1;
     } else {
       this.activeIdx++;
     }
+    this.activeChangedEvent.emit(this.id + '-' + this.activeIdx);
```

Comment:
You've got this `this.activeChangedEvent.emit(this.id + '-' + this.activeIdx)` logic repeated in several places, would be good to extract it. Also, shouldn't `-1` by handled as a special case?